hack/tools/tilt-prepare/main.go

```diff
@@ -287,12 +301,51 @@ func tiltResources(ctx context.Context, ts *tiltSettings) error {
 		if !ok {
 			return errors.Errorf("failed to obtain path for the provider %s", providerName)
 		}
-		tasks[providerName] = providerTask(providerName, fmt.Sprintf("%s/config/default", path), ts)
+		tasks[providerName] = workloadTask(providerName, "provider", "manager", "manager", ts, fmt.Sprintf("%s/config/default", path), getProviderObj)
 	}
 
+	// Add an addon task for each repo defined using the --addon-settings-file flag.
+	for _, f := range *addonSettingsFiles {
+		as, err := readAddonSettings(f)
+		if err != nil {
+			return err
+		}
+		tasks[as.Name] = workloadTask(as.Name, "addon", as.Config.BinaryName, as.Config.ContainerName, ts, fmt.Sprintf("%s/config/default", *as.Config.Context), nil)
+	}
+	// If the test extension addon is enabled add it to the tasks.
+	for _, addon := range ts.EnableAddons {
```

Comment:
Q: That means EnableAddons is only for the test-extension while all others are always enabled?
Could we instead implement this analog to providers?

Reply:
The others are enabled in the tiltfile - i.e. the settings files should only be passed if they're enabled there.

Reply:
We can do it in line with the provider implementation, but this seemed simpler to me.

Reply:
I guess consistency would be nice.

I think having it separated allows us to keep the extension "config" even if we want to disable it (drop it from the enable array).

Having it just once works similarly but then the "config" has to be commented out.

I don't really have strong opinions. Just seems to be nice have it consistent.

I'll take another look at how the config currently would look like on the next review

Reply:
I think now it is analog to providers (?)

Reply:
Yeah - that's the intention of the current implementation.